update/s3.go

```diff
@@ -370,13 +370,61 @@ func promoteRelease(bucketName string, delay time.Duration, hourEastern int, cha
 	return client.PromoteRelease(bucketName, delay, hourEastern, channel, platform, env)
 }
 
+func promoteARelease(releaseName string, bucketName string, platform Platform, env string) (*Release, error) {
+	client, err := NewClient()
+	if err != nil {
+		return nil, err
+	}
+	return client.PromoteDarwinReleaseToProd(releaseName, bucketName, platformDarwin, env)
+}
+
 func updateJSONName(channel string, platformName string, env string) string {
 	if channel == "" {
 		return fmt.Sprintf("update-%s-%s.json", platformName, env)
 	}
 	return fmt.Sprintf("update-%s-%s-%s.json", platformName, env, channel)
 }
 
+// PromoteDarwinReleaseToProd promotes a specific Mac release from test->public
+func (c *Client) PromoteDarwinReleaseToProd(releaseName string, bucketName string, platform Platform, env string) (*Release, error) {
+	if platform.Name != "darwin" {
```

Comment:
Maybe move these checks into promoteARelease and then use platformDarwin and env="prod"  directly?